src/nrnmpi/nrnmpi_dynam.c

```diff
@@ -111,9 +111,16 @@ if (!dlopen("libnrniv.so", RTLD_NOW | RTLD_NOLOAD | RTLD_GLOBAL)) {
 	fprintf(stderr, "Did not promote libnrniv.so to RTLD_GLOBAL: %s\n", dlerror());
 }
 #endif
+#if defined(NRNCMAKE)
```

Comment:
did we need this change? the same is being done as fallback in line 140

Reply:
but note that on line 104, if libmpi is successful and then `libnrnmpi.so` on line 116 will fail and then we return with fatal error.